lib/checks/canFindDenominatorInNumerator.js

```diff
@@ -5,8 +5,21 @@ const Node = require('../node');
 // e.g. (2x+1)/(2x+3) -> True because of the following simplification
 // (2x+1)/(2x+3) -> (2x + 3)/(2x + 3) - 2/(2x + 3) -> 1 - 2/(2x + 3)
 // e.g. (2x+1)/(2x^2 + 3) -> False
+// =========================================================================
+// CHECKS
```

Comment:
I think it would be most clear if you explain what exactly it handles

maybe:

"conditions include being a fraction where the numerator is a sum of ...."

so you see you can include if it's a + node and what the first and second argument is etc into a much more concise language, and that's what I was hoping for.

"Check that the number of arguments in parent node is 2" is already implied in it being a fraction

e.g. several of your conditions can be combined into "the denominator has to be an addition/subtraction of a polynomial term and a constant term"